agent/task.go

```diff
@@ -234,10 +234,14 @@ func (tm *taskManager) run(ctx context.Context) {
 //
 // This used to decide whether or not to propagate a task update to a controller.
 func tasksEqual(a, b *api.Task) bool {
-	a, b = a.Copy(), b.Copy()
+	// save old status and meta
+	aStatus, aMeta, bStatus, bMeta := a.Status, a.Meta, b.Status, b.Meta
 
-	a.Status, b.Status = api.TaskStatus{}, api.TaskStatus{}
-	a.Meta, b.Meta = api.Meta{}, api.Meta{}
+	// set them to the equal value
+	a.Status, a.Meta, b.Status, b.Meta = api.TaskStatus{}, api.Meta{}, api.TaskStatus{}, api.Meta{}
```

Comment:
I'd prever to avoid having `tasksEqual` mutate the tasks. It will limit the ability to use it in concurrent code.

Perhaps we could make this `tasksEqual(a, b api.Task)`. Then blanking out those fields has no effect on the caller. The task object gets copied when the function is called, but it's not a deep copy, so it should be a lot cheaper.